File: Freedom/微信/微信/ChatDetailViewController/WXChatDetailViewController.m
//  FreedomDetailViewController.m
//  Freedom
// Created by Super
#import "WXChatDetailViewController.h"
#import "WXFriendDetailViewController.h"
#import "WXMessageManager.h"
#import <BlocksKit/BlocksKit+UIKit.h>
#import "WXUserGroupCell.h"
#import "WXChatViewController.h"
#import "WXChatFileViewController.h"
#import "WXBgSettingViewController.h"
@interface WXChatDetailViewController () <WechatUserGroupCellDelegate>
@property (nonatomic, strong) WXMessageManager *helper;
@end
@implementation WXChatDetailViewController
- (void)viewDidLoad {
    [super viewDidLoad];
    [self.navigationItem setTitle:@"聊天详情"];
    
    self.helper = [WXMessageManager sharedInstance];
    self.data = [self.helper chatDetailDataByUserInfo:self.user];
    
    [self.tableView registerClass:[WXUserGroupCell class] forCellReuseIdentifier:@"TLUserGroupCell"];
}
#pragma mark - Delegate -
//MARK: UITableViewDataSource
- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    if (indexPath.section == 0 && indexPath.row == 0) {
        WXUserGroupCell *cell = [tableView dequeueReusableCellWithIdentifier:@"TLUserGroupCell"];
        [cell setUsers:[NSMutableArray arrayWithArray:@[self.user]]];
        [cell setDelegate:self];
        return cell;
    }
    return [super tableView:tableView cellForRowAtIndexPath:indexPath];
}
//MARK: UITableViewDelegate
- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    WXSettingItem *item = [self.data[indexPath.section] objectAtIndex:indexPath.row];
    if ([item.title isEqualToString:@"聊天文件"]) {
        WXChatFileViewController *chatFileVC = [[WXChatFileViewController alloc] init];
        [chatFileVC setPartnerID:self.user.userID];
        [self setHidesBottomBarWhenPushed:YES];
        [self.navigationController pushViewController:chatFileVC animated:YES];
    }else if ([item.title isEqualToString:@"设置当前聊天背景"]) {
        WXBgSettingViewController *chatBGSettingVC = [[WXBgSettingViewController alloc] init];
        [chatBGSettingVC setPartnerID:self.user.userID];
        [self setHidesBottomBarWhenPushed:YES];
        [self.navigationController pushViewController:chatBGSettingVC animated:YES];
    }else if ([item.title isEqualToString:@"清空聊天记录"]) {
        [self showAlerWithtitle:nil message:nil style:UIAlertControllerStyleActionSheet ac1:^UIAlertAction *{
            return [UIAlertAction actionWithTitle:@"清空聊天记录" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
                BOOL ok = [[WXMessageManager sharedInstance] deleteMessagesByPartnerID:self.user.userID];
                if (!ok) {
                    [SVProgressHUD showErrorWithStatus:@"清空聊天记录失败"];
                }else{
                    [[WXChatViewController sharedChatVC] resetChatVC];
                }
            }];
        } ac2:^UIAlertAction *{
            return [UIAlertAction actionWithTitle:@"取消" style:UIAlertActionStyleCancel handler:^(UIAlertAction * _Nonnull action) {
            }];
        } ac3:nil completion:nil];
    }
    [self.tableView deselectRowAtIndexPath:indexPath animated:NO];
}
- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath{
    if (indexPath.section == 0 && indexPath.row == 0) {
        NSUInteger count = self.user ? 1 : 0;
        return ((count + 1) / 4 + ((((count + 1) % 4) == 0) ? 0 : 1)) * 90 + 15;
    }
    return [super tableView:tableView heightForRowAtIndexPath:indexPath];
}
//MARK: TLUserGroupCellDelegate
- (void)userGroupCellDidSelectUser:(WXUser *)user{
    WXFriendDetailViewController *detailVC = [[WXFriendDetailViewController alloc] init];
    [detailVC setUser:user];
    [self setHidesBottomBarWhenPushed:YES];
    [self.navigationController pushViewController:detailVC animated:YES];
}
- (void)userGroupCellAddUserButtonDown{
    [SVProgressHUD showInfoWithStatus:@"添加讨论组成员"];
}
@end

File: Freedom/微信/微信/ChatDetailViewController/WXChatDetailViewController.swift
//
//  WXChatDetailViewController.swift
//  Freedom

import Foundation
class WXChatDetailViewController: WXSettingViewController, WechatUserGroupCellDelegate {
    var user: WXUser = WXUser()
    var helper = WXMessageHelper.shared
    override func viewDidLoad() {
        super.viewDidLoad()
        navigationItem.title = "聊天详情"
        data = helper.chatDetailData(byUserInfo: user)
        tableView.register(WXUserGroupCell.self, forCellReuseIdentifier: WXUserGroupCell.identifier)
    }
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        if indexPath.section == 0 && indexPath.row == 0 {
            let cell = tableView.dequeueCell(WXUserGroupCell.self)
            cell.users = [user]
            cell.delegate = self
            return cell
        }
        return super.tableView(tableView, cellForRowAt: indexPath)
    }
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
//        let item = data[indexPath.section].items[indexPath.row]
//        if (item.title == "聊天文件") {
//            let chatFileVC = WXChatFileViewController()
//            chatFileVC.partnerID = user.userID
//            navigationController.pushViewController(chatFileVC, animated: true)
//        } else if (item.title == "设置当前聊天背景") {
//            let chatBGSettingVC = WXBgSettingViewController()
//            chatBGSettingVC.partnerID = user.userID
//            navigationController.pushViewController(chatBGSettingVC, animated: true)
//        } else if (item.title == "清空聊天记录") {
//            showAlerWithtitle(nil, message: nil, style: UIAlertController.Style.actionSheet, ac1: {
//                return UIAlertAction(title: "清空聊天记录", style: .default, handler: { action in
//                    let ok = WXMessageManager.shared.deleteMessages(byPartnerID: self.user.userID)
//                    if !ok {
//                        noticeError("清空聊天记录失败")
//                    } else {
//                        WXChatViewController.shared.resetChatVC()
//                    }
//                })
//            }, ac2: {
//                return UIAlertAction(title: "取消", style: .cancel, handler: { action in
//                })
//            }, ac3: nil, completion: nil)
//        }
        self.tableView.deselectRow(at: indexPath, animated: false)
    }
    override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        if indexPath.section == 0 && indexPath.row == 0 {
            let count: Int = 1
            return CGFloat(((count + 1) / 4 + ((((count + 1) % 4) == 0) ? 0 : 1)) * 90 + 15)
        }
        return super.tableView(tableView, heightForRowAt: indexPath)
    }
    func userGroupCellDidSelect(_ user: WXUser) {
        let detailVC = WXFriendDetailViewController()
        detailVC.user = user
        navigationController?.pushViewController(detailVC, animated: true)
    }
    func userGroupCellAddUserButtonDown() {
        noticeInfo("添加讨论组成员")
    }
}
